src/webserver/DOSGuard.h

```diff
@@ -9,11 +9,48 @@
 class DOSGuard
 {
     boost::asio::io_context& ctx_;
-    std::mutex mtx_;  // protects ipFetchCount_
+    std::recursive_mutex
```

Comment:
Not in the scope for this PR, but we should consider how to do more granular locking for the DOSGuard.

Reply:
Actually, let's try to do that here. I think we can switch the values of the map to atomic, and then switch this to a shared_mutex. Then, we only need a unique lock when we are adding to one of the maps. @ethanlabelle